tests/agent_disconnect_test.go

```diff
@@ -0,0 +1,151 @@
+package tests
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client"
+)
+
+func TestProxy_Agent_Disconnect_Same_Stream(t *testing.T) {
```

Comment:
I think the second test covers the first test. If so, let's remove this test.

Reply:
The second test actually tests the disconnect scenario with 2 requests using 2 different tunnels. This first test uses HTTP 1.1 persistent connections to send 2 requests on the same tunnel. k/k has some commands using the latter scenario so I think it would be beneficial to keep both cases.

I will reword this test case a bit more accurately.

Reply:
Got it. Can you also say this in the comments of the tests?